content/blog/bare-metal-provisioning-on-hpe-greenlake-using-terraform.md

```diff
@@ -0,0 +1,353 @@
+
+In this blog post, I will walk you through the steps required to use the HPE GreenLake Terraform Provider to deploy and further manage bare metal compute instances.
+
+# Preparing an Infrastructure-as-Code implementation
+
+## Setting up an API client for access
+
+You need an API client to authenticate against HPE GreenLake.
+
+Follow the below steps for an API client creation:
+
+1. From the HPE GreenLake platform, launch the HPE GreenLake Central console for the appropriate tenant. Under the settings icon on the tenant Dashboard page, select the **User Management** option. 
+
+![User Management](/img/greenlake_console_usermanagement.png "User Management")
+
+2. Under the API Clients tab, click on **Create API Client**.
+
+![Create API Client](/img/greenlake_console_createapiclient.png "Create API Client")
+
+3. Enter a **Name** (mandatory field) and **Description** (optional) for the API client, and click on **Create** button.
+
+![Create API Client](/img/greenlake_console_createapiclient_form.png "Create API Client")
+
+4. Ensure you make a note of the **Issuer**, **Client ID**, and **Client Secret** before clicking on the **Close** button. These details will be exported as environment variables in the next section.
+
+![API Client Created](/img/greenlake_conolse_apiclient_created.png "API Client Created")
+
+5. In the API Clients page, select the newly created client, and click on **Create Assignment** button.
+6. Assign the roles **BMAAS Access Viewer** and **BMAAS Access Project Contributor** on the **Space: Default.**
+
+![BMaaS Roles](/img/greenlake_console_createbmaasassignment.png "BMaaS Roles")
+
+The API client is now ready to be used to run the Terraform resources.
+
+## Select the Compute group ID
+
+A compute group is a logical grouping of bare metal resources (compute instances, networks, SSH keys, etc.) that a team of cloud consumers can consume. You must specify the compute group ID to interact with the bare metal service resources. You can create a new compute group or select the existing one from the HPE GreenLake console. Make a note of the compute group ID because you need it to set a variable.
+
+***Note: Compute group is an AKA Project.***
+
+1. Navigate to HPE GreenLake for Private Cloud Services card -> Bare Metal -> Compute Groups.
+
+![BMaaS Compute Groups](/img/compute_group_list.png "BMaaS Compute Groups")
+
+2. Click on the desired compute group and then extract the ID from the browser URL seen at that time.  
+
+   This will be exported in the environment variable **HPE_METAL_PROJECT_ID** in the later section.
+
+![Compute group ID](/img/compute_group_id.png "Compute group ID")
+
+## Install Terraform
+
+Next, get your system ready to run Terraform. In case this has not been done yet:
+
+1. Download and install Terraform, version v0.13 or later.
+   For more information, see [https://learn.hashicorp.com/tutorials/terraform/install-cli](<1. https://learn.hashicorp.com/tutorials/terraform/install-cli>). 
```

Comment:
For more information, see the [Terraform Installation guide](https://learn.hashicorp.com/tutorials/terraform/install-cli)